File: codeforces/div4/E/main.py
for _ in range(int(input())):
    n,m,k=map(int,input().split())
    n_bigger = n > m
    if abs(n-m) > k or k > max(n,m):
        print(-1)
        continue
    s = ''
    for __ in range(k):
        s += '0' if n_bigger else '1'
        if n_bigger: n-=1
        else: m-=1
    while n > 0 or m > 0:
        if n > m or (s[-1] == '1' and n == m):
            s += '0'
            n-=1
        else:
            s += '1'
            m-=1
    print(s)

File: codeforces/div4/E/main.cpp
#include <bits/stdc++.h>
using namespace std;

#define rep(i, a, b) for(int i = a; i < (b); ++i)
#define all(x) begin(x), end(x)
#define sz(x) (int)(x).size()
typedef long long ll;
typedef pair<int, int> pii;
typedef vector<int> vi;

int main() {
    cin.tie(0)->sync_with_stdio(0);
    //cin.exceptions(cin.failbit);
    int T; cin >> T;
    rep(t, 0, T) {
        ll n, m, k; cin >> n >> m >> k;
        bool n_bigger = n > m;
        if (abs(n-m) > k || k > max(n,m)) {
            printf("-1\n");
            continue;
        }
        string s = "";
        rep(i, 0, k) {
            s.push_back(n_bigger ? '0' : '1');
            if (n_bigger) n--;
            else m--;
        }
        while (n > 0 || m > 0) {
            if (n > m || (s.back() == '1' && n == m)) {
                s.push_back('0');
                n--;
            } else {
                s.push_back('1');
                m--;
            }
        }
        printf("%s\n", s.c_str());
    }
}